test/parallel/test-repl-.save.load.js

```diff
@@ -74,5 +74,20 @@ putIn.write = function(data) {
 };
 putIn.run(['.load ' + loadFile]);
 
+// clear the REPL
+putIn.run(['.clear']);
+
+// NUL (\0) is disallowed in filenames in UNIX-like operating systems and 
+// Windows so we can use that to test failed saves
+var invalidFileName = join(common.tmpDir, '\0\0\0\0\0');
 
-//TODO how do I do a failed .save test?
+// should not break
+putIn.write = function(data) {
+  // make sure I get a failed to save message and not some other error
+  assert.equal(data, 'Failed to save:' + invalidFileName + '\n');
+  // reset to no-op
+  putIn.write = function() {};
```

Comment:
It might be clearer and more robust to override `fs.writeFileSync()` or whatever. Downside is that ties the test to the REPL implementation a little more closely than one might like.